src/nvim/window.c

```diff
@@ -4745,7 +4745,9 @@ void win_new_width(win_T *wp, int width)
   wp->w_redr_status = TRUE;
 
   if (wp->w_buffer->terminal) {
-    terminal_resize(wp->w_buffer->terminal, wp->w_width, 0);
+    if (wp->w_height != 0) {
+      terminal_resize(wp->w_buffer->terminal, wp->w_width, 0);
```

Comment:
Actually those parameters to terminal_resize are useless. Terminal_resize will loop over all windows on the terminal-buffer to figure out what the minimum size is, and use that as the new size. Passing in the value for this specific window adds no value.

The other PR already contains a commit which removes those unnecessary parameters.